core/services/relay/evm/evm_service.go

```diff
@@ -128,6 +134,94 @@ func (r *Relayer) GetTransactionStatus(ctx context.Context, transactionID common
 	return commontypes.TransactionStatus(status), nil
 }
 
+func (r *Relayer) SubmitTransaction(ctx context.Context, txRequest evmtypes.SubmitTransactionRequest) (*evmtypes.TransactionResult, error) {
+	config := r.chain.Config()
+
+	fromAddress := config.EVM().Workflow().FromAddress().Address()
+	var gasLimit uint64 = 0
+	if txRequest.GasConfig != nil && txRequest.GasConfig.GasLimit != nil {
+		gasLimit = *txRequest.GasConfig.GasLimit
+	}
+
+	uuid, err := uuid.NewUUID()
+	if err != nil {
+		return nil, err
+	}
+	txID := uuid.String()
+
+	// PLEX-1524 - review which transmitter checker we should use
+	var checker evmtxmgr.TransmitCheckerSpec
+	checker.CheckerType = evmtxmgr.TransmitCheckerTypeSimulate
+	value := big.NewInt(0)
+
+	// PLEX-1524 - Define how we should properly get the workflow execution ID into the meta without making the API CRE specific.
+	var txMeta *txmgrtypes.TxMeta[common.Address, common.Hash]
+	txmReq := evmtxmgr.TxRequest{
+		FromAddress:    fromAddress,
+		ToAddress:      txRequest.To,
+		EncodedPayload: txRequest.Data,
+		FeeLimit:       gasLimit,
+		Meta:           txMeta,
+		IdempotencyKey: &txID,
+		// PLEX-1524 - Review strategy to be used.
+		Strategy:       txmgr.NewSendEveryStrategy(),
+		Checker:        checker,
+		Value:          *value,
+	}
+
+	_, err = r.chain.TxManager().CreateTransaction(ctx, txmReq)
+	if err != nil {
+		return nil, fmt.Errorf("%w; failed to create tx", err)
+	}
+
+
+	maximumWaitTimeForConfirmation := config.EVM().ConfirmationTimeout()
+	start := time.Now()
+	for {
+		txStatus, err := r.chain.TxManager().GetTransactionStatus(ctx, txID)
+		if err != nil {
+			return nil, err
+		}
+		if txStatus == commontypes.Fatal {
+			return &evmtypes.TransactionResult{
+				TxStatus: evm.TxFatal,
+				TxHash:   evmtypes.Hash{},
+			}, nil
+		}
+		if txStatus == commontypes.Unconfirmed || txStatus == commontypes.Finalized {
+			break
+		}
+		if time.Since(start) > maximumWaitTimeForConfirmation {
+			return nil, errors.Errorf("Wait time for Tx %s to get confirmed was greater than maximum wait time %d", txID, maximumWaitTimeForConfirmation)
+		}
+		//PLEX-1524 - Use ticker instead of time.Sleep and make the time configurable
+		time.Sleep(100 * time.Millisecond)
+	}
+
+	receipt, err := r.chain.TxManager().GetTransactionReceipt(ctx, txID)
+	if err != nil {
+		return nil, fmt.Errorf("%w; failed to get TX receipt for tx with ID %s", err, txID)
```

Comment:
It's more common to add errors as suffix
```suggestion
		return nil, fmt.Errorf("failed to get TX receipt for tx with ID %s: %w", txID, err)
```

Reply:
Done